openadmet/models/anvil/workflow.py

```diff
@@ -310,7 +309,13 @@ def run(
         logger.info("Model saved")
 
         logger.info("Predicting")
-        y_pred = self.model.predict(X_test_feat)
+        # Check if the model has predict_proba method (classification)
+        if hasattr(self.model, "predict_proba"):
```

Comment:
Works but another option is to use a class attribute of the model to mark this e.g `model.is_classifier` or something like that. 

`hasattr` is ok, just would be cleaner with an API touchpoint IMO as only sklearn compatible will have `predict_proba`